### PR TITLE
feat(RadioGroup): improve RadioGroup component

### PR DESCRIPTION
**Pull request by beawar**
Replace div with fieldset.
Make type of value generic, so that it can be reduced to a subset of type string.

refs: CDS-167 CDS-57

### CODE REVIEW

File: src/components/inputs/RadioGroup.test.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+			</RadioGroup>
+		);
+		expect(screen.getByRole('radio', { name: 'item 1' })).not.toBeChecked();
+		expect(screen.getByRole('radio', { name: 'item 2' })).toBeChecked();
+		expect(screen.getByRole('radio', { name: 'item 3' })).not.toBeChecked();
+	});
+
+	test('should check the radio which match the value', () => {
+		render(
+			<RadioGroup defaultValue={'r2'} value={'r3'}>
```
Comment (CataldoMazzilli) [inline src/components/inputs/RadioGroup.test.tsx:44]: remove defaultValue prop?

File: src/components/inputs/RadioGroup.test.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+		);
+		userEvent.click(screen.getByRole('radio', { name: 'item 2' }));
+		expect(onClick1Fn).not.toHaveBeenCalled();
+		expect(onClick2Fn).toHaveBeenCalled();
+		expect(onClick3Fn).not.toHaveBeenCalled();
+	});
+
+	test('should set all radio as disabled when the group is disabled', () => {
+		render(
+			<RadioGroup defaultValue={'r2'} value={'r3'} disabled>
```
Comment (CataldoMazzilli) [inline src/components/inputs/RadioGroup.test.tsx]: remove defaultValue?

**Review by CataldoMazzilli — DISMISSED** [review]

File: src/components/inputs/RadioGroup.test.tsx (hunk 1 above)
Comment (beawar) [reply src/components/inputs/RadioGroup.test.tsx:44]: Here I left it to check that the value prop wins over the defaultValue. I set two different values in the two intentionally, to be sure that the r3 is selected, and not the r2 of the default value

File: src/components/inputs/RadioGroup.test.tsx (hunk 2 above)
Comment (beawar) [reply src/components/inputs/RadioGroup.test.tsx]: Yep this is a copy and paste leftover

**Review by nubsthead — DISMISSED** [review]

**Review by CerDennis — DISMISSED** [review]

**Review by nubsthead — APPROVED** [review]

**Review by CataldoMazzilli — APPROVED** [review]

**Pull request merged**